engine/src/chunk.cpp

```diff
@@ -4242,6 +4242,12 @@ bool MCChunk::getprop(MCExecContext& ctxt, Properties which, MCNameRef index, Bo
         if (islinechunk() && t_info == nil)
```

Comment:
I wonder if we can simplify this code a bit - we are calling lookup_object_property (conditionally) 3 times which seems a bit overkill :)

What we want to achieve is this:

If the chunk is a line chunk, then lookup the line property first, then lookup char property if none found. (This is correct as a line chunk is always a char chunk - we need this rule as there are some properties which are both line and char level).

If the chunk is not a line chunk, then lookup the char property first, then lookup line property if none found. (This is correct as it is usual practice to be able to set a line property on a char and have it apply to the whole line - think things like text alignment in word-processors).

Thus the code could be reworked to be:

```
t_info = lookup_object_property(..., islinechunk() ? kMCPropertyInfoChunkTypeLine : kMCPropertyInfoChunkTypeChar);
if (t_info == nil)
  t_info = lookup_object_property(..., islinechunk() ? kMCPropertyInfoChunkTypeChar : kMCPropertyInfoChunkTypeLine);
```